### ISSUE TITLE
Incorrect dependencies for package

### ISSUE THREAD

**Issue by jeromekelleher** [tskit-dev/tsinfer#938]
The just shipped version 0.3.2 has these dependencies in pyproject.toml:
```
   "numpy>=1.23.5",
    "six",
    "tqdm",
    "humanize",
    "daiquiri",
    "tskit>=0.5.3",
    "numcodecs>=0.6",
    "zarr>=2.2,!=2.11.0,!=2.11.1,!=2.11.2,<3",
    "lmdb",
    "sortedcontainers",
    "attrs>=19.2.0",
    "dask[array]",
    "numba",
```

We certainly don't need six, and I thought that we had removed any usage of dask. I'm not clear why we have a numba dependency either - we're not using numba for any actual production code in tsinfer yet, are we?

We probably need to ship another release to fix this, as these dependencies will cause problems for installation.

**Comment by benjeffery** [comment]
I missed this when back porting. 0.4 will need numba, and dask isn't removed in `main` yet. Will fix this morning.

**Comment by benjeffery** [comment]
Fixed in 0.3.3
